R/pkg/R/mllib.R

```diff
@@ -126,3 +126,27 @@ setMethod("summary", signature(object = "PipelineModel"),
               return(list(coefficients = coefficients))
             }
           })
+
+#' Fit a k-means model
+#'
+#' Fit a k-means model, similarly to R's kmeans().
+#'
+#' @param x DataFrame for training
+#' @param centers Number of centers
+#' @param iter.max Maximum iteration number
+#' @param algorithm Algorithm choosen to fit the model
+#' @return A fitted k-means model
+#' @rdname kmeans
+#' @export
+#' @examples
+#'\dontrun{
+#' model <- kmeans(x, centers = 2, algorithm="random")
+#'}
+setMethod("kmeans", signature(x = "DataFrame"),
```

Comment:
I don't think we can make the signature the same as stats:kmeans for now, since `KMeans` in MLlib doesn't provide the same interface with R's. E.g. `KMeans` in MLlib doesn't support warm-start so the `centers` has only to be a "numeric", and it doesn't have arguments like `nstart` and `trace`. Refer to [the implementation of `glm` in SparkR](https://github.com/apache/spark/blob/master/R/pkg/R/mllib.R#L53), which also cannot match the signature with R's 100%.